src/hope_parser.h

```diff
@@ -132,6 +136,9 @@ class Parser {
     void finishPatch(ParseNode pn) noexcept;
     #endif
 
+    void import(ParseNode pn);
+
+    //DO THIS: the parser should not assume there is a single Typeset::Model
     std::vector<std::pair<size_t, size_t>> token_map_stack;
     std::vector<size_t> token_stack_frames;
     const std::vector<Token>& tokens;
```

Comment:
The namespaces are good, but the assumption that a program corresponds to a single file is deeply entrenched- and that's not even thinking about good editor interaction for multiple files. It may be worth cleaning up the namespace changes and merging before tackling file support.